larray/core.py

```diff
@@ -7724,12 +7724,12 @@ def read_csv(filepath, nb_index=0, index_col=[], sep=',', headersep=None,
             pos_last = 0 if dialect != 'liam2' else len(header) - 1
         axes_names = header[:pos_last + 1]
 
-    if len(index_col) == 0 and nb_index == 0:
+    if index_col is None and nb_index == 0:
         nb_index = len(axes_names)
         if dialect == 'liam2':
             nb_index -= 1
 
-    if len(index_col) > 0:
+    if isinstance(index_col, list) and len(index_col) > 0:
```

Comment:
len(index_col) > 0 is not longer necessary I think (it was to differentiate from the default value)